pandas/core/generic.py

```diff
@@ -8680,24 +8682,304 @@ def _doc_parms(cls):
 """
 
 _cnum_doc = """
+Return %(desc)s over a DataFrame or Series axis.
+
+Returns a DataFrame or Series of the same size containing the %(desc)s.
 
 Parameters
 ----------
-axis : %(axis_descr)s
+axis : {0 or 'index', 1 or 'columns'}, default 0
+    The index or the name of the axis. 0 is equivalent to None or 'index'.
 skipna : boolean, default True
     Exclude NA/null values. If an entire row/column is NA, the result
-    will be NA
+    will be NA.
+*args, **kwargs
 
 Returns
 -------
-%(outname)s : %(name1)s\n
-
-
+%(outname)s : %(name1)s or %(name2)s\n
+%(examples)s
 See also
 --------
 pandas.core.window.Expanding.%(accum_func_name)s : Similar functionality
     but ignores ``NaN`` values.
+Series.%(outname)s : Return %(desc)s over Series axis.
+DataFrame.%(accum_func_name)s : Return the %(accum_func_name)s over
+    DataFrame axis.
```

Comment:
@jorisvandenbossche I used templating and `%(accum_func_name)s` to add `DataFrame.prod` to 'See also' ection of `DataFrame.cumprod`, etc. All methods look good except `DataFrame.prod` where it reads: 'Return the prod over DataFrame axis' instead of 'Return the product over...'. Any suggestions on this? Should we even worry about this?

Reply:
One relatively easy solution would be to change the ``desc`` param to `_make_cum_function` from "cumulative produce" to "product", because we can put the "cumulative" (which is the same for each of the 4 functions" in the template itself. Then you can use that name in the see also description.